### ISSUE TITLE
Add http2 support

### ISSUE THREAD

**Issue by doug-wade** [koajs/koala#48]
Http2 has landed in node.js.  We should add support.

**Comment by nickmccurdy** [comment]
The API level for it is 1 (experimental), we might want to wait until this changes to put it in a stable release. Also, does this replace #16 or should we still fix that first?

**Comment by doug-wade** [comment]
I don't think we should wait, I think we should add the feature now but make sure its completely optional, so clients can test and develop against the http2 api sooner.

**Comment by nickmccurdy** [comment]
Good point, though I don't think we should make this block any releases until it's no longer experimental.
